src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.interfaces.UserDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to add user", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public User findByEmail(String email) {
```

Comment:
it could return the Optional
```suggestion
    public Optional<User> findByEmail(String email) {
```

query.uniqueResult(); can return null if there is no user, so optional is applicable is such case